Deprecate the routes only version of the config

cah-briangantzler opened the pull request.

cah-briangantzler commented:
So use to have prettier run when I commit things, forgot to run it separately here. 

Mind if I add lint-staged and run prettier then lint on commit. Commits will fail to commit if lint fails, which is good, and prettier will pretty it up so lint wont fail on prettier problems.

SergeAstapov commented on tests/dummy/app/pods/docs/advanced/server-configuration/template.md:
```diff
@@ -1,22 +1,13 @@
 # Server configuration
 
-The MirageJS server is configured for you by ember-cli-mirage. However, if you
```
@cah-briangantzler minor, thinking if would be more helpful to say `The MirageJS server used to be configured for you by ember-cli-mirage.`

SergeAstapov commented on tests/dummy/app/pods/docs/advanced/server-configuration/template.md:
```diff
@@ -1,22 +1,13 @@
-then creating a new default export function that creates the mirageJS server.
-
-This new default export function receives a single argument named `config`, 
+The MirageJS server has been configured for you by ember-cli-mirage. This hase been
```
"This hase been" -> "This has been"

SergeAstapov commented:
> So use to have prettier run when I commit things, forgot to run it separately here.
> 
> Mind if I add lint-staged and run prettier then lint on commit. Commits will fail to commit if lint fails, which is good, and prettier will pretty it up so lint wont fail on prettier problems.

sounds good!
Mind to do such setup in separate PR?

SergeAstapov commented:
@cah-briangantzler CI should be green as of #2327, mind to rebase onto master?

cah-briangantzler commented:
I thought I was up to date. My bad I didnt check. Was wondering about that error showing up again. Repushed, waiting on checks.

Yea of course a separate PR

cah-briangantzler commented:
Updated the docs, then rebased onto master.Then remembered, this is a deprecation and shouldnt be on master, it should be on v2 shouldnt it?

I updated the docs for V2. All these docs need re-written again with V3 and there be a migration section from V2 to V3.   


I think I need to rebase this back to V2. Were the checks green on the V2 branch ?

cah-briangantzler commented:
closing and starting over, history is muddled
